include/swift/AST/ClangModuleLoader.h

```diff
@@ -40,6 +43,66 @@ enum class ClangTypeKind {
   ObjCProtocol,
 };
 
+/// A path for serializing a declaration.
+class StableSerializationPath {
+public:
+  struct ExternalPath {
+    enum StepKind {
```

Comment:
What do you think about renaming this `ComponentKind`? That's the typical terminology I've seen around filepaths.